docs/tutorials/forum-application/3.send-message.md

```diff
@@ -0,0 +1,381 @@
+an invalid transaction format.
+
+Then it, searches for a user in the database using the `app.state.DB.FindUserByName` function. If the user is not found,
+it prints a message indicating that the user was not found. If there is an error other than a key not found error,
+it prints an error message and returns a response with an error code indicating an encoding error.
+
+If the user is found and is not banned, it returns a response with a success code (`CodeTypeOK` which in CometBFT a `0`
+code value to indicate success).
+
+Finally, it prints a success message indicating the success of the check transaction.
+
+*Tip: The function `CheckTx` is a stateless function that is primarily used by the application to check if the a tx is
+valid or not as per the application criteria (well-formed and from a valid user)*
+
+Note: You will learn about different packages and functions like `app.state.DB.FindUserByName` in the upcoming sections.
+In this section you will learn about the ABCI methods in the `ForumApp`.
+
+## PrepareProposal
+
+The `PrepareProposal` method is responsible for creating the contents of the proposed block, typically by selecting a
+set of transactions that should be included in the next block. It may use various criteria to determine which transactions
+to include, such as transaction fees, priority, or application-specific rules (as defined by the application).
+
+The method takes in a `PrepareProposalRequest` object, which contains information about the current state of the blockchain,
+such as the current height and the last committed block hash. It may also include other relevant information that the
+application needs to generate the proposal.
+
+The `PrepareProposal` method should return a `PrepareProposalResponse` object, which includes the proposed block contents.
+This typically includes the list of transactions (txs) that should be included in the next block.
+
+**Following is the code for the `PrepareProposal` method:**
+
+```go
+// PrepareProposal is used to prepare a proposal for the next block in the blockchain. The application can re-order, remove
+// or add transactions
+func (app *ForumApp) PrepareProposal(_ context.Context, req *abci.PrepareProposalRequest) (*abci.PrepareProposalResponse, error) {
+    fmt.Println("Executing Application PrepareProposal")
+
+	// Get the curse words from the vote extensions
+	voteExtensionCurseWords := app.getWordsFromVe(req.LocalLastCommit.Votes)
+
+	// Prepare req puts the BanTx first, then adds the other transactions
+	// ProcessProposal should verify this
+	proposedTxs := make([][]byte, 0)
+	finalProposal := make([][]byte, 0)
+	bannedUsersString := make(map[string]struct{})
+	for _, tx := range req.Txs {
+		msg, err := model.ParseMessage(tx)
+		if err != nil {
+			continue
+		}
+		// Adding the curse words from vote extensions too
+		if !hasCurseWord(msg.Message, voteExtensionCurseWords) {
+			proposedTxs = append(proposedTxs, tx)
```

Comment:
If you add a `continue` here, you can get rid of the `else` and all its indentation, which makes the code hard to follow